src/term.rs

```diff
@@ -259,6 +266,14 @@ impl BufWrite for Attrs {
             }
         }
 
+        if let Some(dim) = self.dim {
+            if dim {
+                write_param!(2);
+            } else {
+                write_param!(22);
```

Comment:
this isn't correct - text that is only bold will end up no longer being bold after this, because turning off dim also turns off bold. you'll probably need to handle these cases together to make sure all of the different potential settings work together properly.